aiida/engine/processes/functions.py

```diff
@@ -234,10 +234,14 @@ def build(func: Callable[..., Any], node_class: Type['ProcessNode']) -> Type['Fu
 
         def _define(cls, spec):  # pylint: disable=unused-argument
             """Define the spec dynamically"""
+            from plumpy.ports import UNSPECIFIED
+
             super().define(spec)
 
             for i, arg in enumerate(args):
-                default = ()
+
+                default = UNSPECIFIED
+
                 if defaults and i >= first_default_pos:
                     default = defaults[i - first_default_pos]
```

Comment:
Do we need these lines?  It seems that they are always skipped.

Reply:
Why do you think that? If we have the following function signature:
```python
def function(a, b, c, d=1, e=None, f=True)
```
we have the following variables
```
nargs = 6
ndefaults = 3
first_default_pos = 3
```
and so when we iterate over `args` the condition `if defaults and i>= first_default_pos` evaluates to `True` once `i` is greater equal than `3`, at which point the default is taken from the `defaults` array, which is `[1, None, True]` in this example. In all other cases, namely the first arguments without defaults, the default is `UNSPECIFIED`.

I think this is correct no?  Or am I missing something?

Reply:
Thanks. Yes, you are right.